sources/models/src/modeled_types/kubernetes.rs

```diff
@@ -437,3 +437,278 @@ mod test_kubernetes_bootstrap_token {
+    ImagefsAvailable,
+    #[serde(rename = "imagefs.inodesFree")]
+    ImagefsInodesFree,
+    #[serde(rename = "pid.available")]
+    PidAvailable,
+}
+
+impl TryFrom<&str> for EvictionHardKey {
+    type Error = error::Error;
+
+    fn try_from(input: &str) -> Result<Self, Self::Error> {
+        serde_plain::from_str::<EvitionSignal>(&input).context(error::InvalidPlainValue {
+            field: "Eviction Hard key",
+        })?;
+        Ok(EvictionHardKey {
+            inner: input.to_string(),
+        })
+    }
+}
+string_impls_for!(EvictionHardKey, "EvictionHardKey");
+
+#[cfg(test)]
+mod test_kubernetes_eviction_hard_key {
+    use super::EvictionHardKey;
+    use std::convert::TryFrom;
+
+    #[test]
+    fn good_eviction_hard_key() {
+        for ok in &[
+            "memory.available",
+            "nodefs.available",
+            "nodefs.inodesFree",
+            "imagefs.available",
+            "imagefs.inodesFree",
+            "pid.available",
+            ] {
+            EvictionHardKey::try_from(*ok).unwrap();
+        }
+    }
+
+    #[test]
+    fn bad_eviction_hard_key() {
+        for err in &["", "storage.available", ".bad", "bad.", &"a".repeat(64)] {
+            EvictionHardKey::try_from(*err).unwrap_err();
+        }
+    }
+}
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// KubernetesThresholdValue represents a string that contains a valid kubernetes threshold value.
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct KubernetesThresholdValue {
+    inner: String,
+}
+
+lazy_static! {
+    pub(crate) static ref KUBERNETES_QUANTITY: Regex = Regex::new(r"^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$").unwrap();
+}
+
+impl TryFrom<&str> for KubernetesThresholdValue {
+    type Error = error::Error;
+
+    fn try_from(input: &str) -> Result<Self, Self::Error> {
+        if input.ends_with("%") {
+            let input_f32 = input.replace("%", "").parse::<f32>().context(error::ParseToF32 { input })?;
+            ensure!(
+                (0.0..100.0).contains(&input_f32),
+                error::InvalidKubernetesQuantity { input }
+            )
+        }
+        else{
+            ensure!(
+                KUBERNETES_QUANTITY.is_match(input),
+                error::InvalidKubernetesQuantity { input }
+            );
+        };
+
+        Ok(KubernetesThresholdValue{
+            inner: input.to_string(),
+        })
+    }
+}
+string_impls_for!(KubernetesThresholdValue, "KubernetesThresholdValue");
+
+#[cfg(test)]
+mod test_kubernetes_threshold_value {
+    use super::KubernetesThresholdValue;
+    use std::convert::TryFrom;
+
+    #[test]
+    fn good_kubernetes_threshold_value() {
+        for ok in &[ "10%", "10Mi", "1024M", "1Gi" ] {
+            KubernetesThresholdValue::try_from(*ok).unwrap();
+        }
+    }
+
+    #[test]
+    fn bad_kubernetes_threshold_value() {
+        for err in &["", "anything%", ".bad", "bad.", &"a".repeat(64)] {
```

Comment:
Looks like you need to run `cargo fmt`. There are also some other places that need formatting.